### PR TITLE
elfparser: Add method to load BPF from an io.ReaderAt

### PR DESCRIPTION
**Pull request by shun159**
This patch adds the `LoadBpfFromReader` to the `bpfSDKClient` to load an ELF file from an `io.ReaderAt`. With this patch, it becomes possible to load an ELF binary embedded in an application binary using `go:embed` as the following:

```go
package main

import (
	"bytes"
	_ "embed"

	"github.com/aws/aws-ebpf-sdk-go/pkg/elfparser"
)

func main() {
	elfc := elfparser.New()
	reader := bytes.NewReader(_ElfFileBytes)
	elfc.LoadBpfFromReader(reader, "global")
}

//go:embed bpf/kprobe.bpf.elf
var _ElfFileBytes []byte
```

### CODE REVIEW

**Comment by dims** [comment]
I like it. do you want to add a test too?

File: pkg/elfparser/elf.go [hunk 1]
```diff
@@ -122,26 +122,35 @@ func newElfLoader(elfFile *elf.File, bpfmapapi ebpf_maps.BpfMapAPIs, bpfprogapi
 	return elfloader
 }
 
-func (b *bpfSDKClient) LoadBpfFile(path, customizedPinPath string) (map[string]BpfData, map[string]ebpf_maps.BpfMap, error) {
-	bpfFile, err := os.Open(path)
+// LoadBpfFromReader loads ELF file from io.ReaderAt and parses an ELF file into BPF objects
+func (b *bpfSDKClient) LoadBpfFromReader(r io.ReaderAt, customizedPinPath string) (
```
Comment (jayanthvn) [inline pkg/elfparser/elf.go:126]: You need to add it to the - interface `type BpfSDKClient `

Comment (shun159) [reply pkg/elfparser/elf.go:127]: Oh! Thank you for pointing that out. 
Indeed, we needed to add the interface. I've fixed to add.

**Comment by shun159** [comment]
Hi @dims, Thank you for your review.
Sure! I've added the testcase.

**Comment by shun159** [comment]
I've made fixes based on the feedback. Happy with these changes?

**Pull request closed without merging**